src/commands/protocols/common/epoch-check/blockchain-epoch-check-command.js

```diff
@@ -24,76 +24,74 @@ class EpochCheckCommand extends Command {
     }
 
     async execute(command) {
-        this.logger.info('Epoch check: Starting epoch check command');
-        const operationId = this.operationIdService.generateId();
-
-        await Promise.all(
-            this.blockchainModuleManager.getImplementationNames().map(async (blockchain) => {
-                this.operationIdService.emitChangeEvent(
-                    OPERATION_ID_STATUS.COMMIT_PROOF.EPOCH_CHECK_START,
-                    operationId,
-                    blockchain,
-                );
+        const { operationId, blockchain } = command.data;
+        this.logger.info(
+            `Epoch check: Starting blockchain epoch check command for ${blockchain} with operation id: ${operationId}`,
+        );
 
-                const commitWindowDurationPerc =
-                    await this.blockchainModuleManager.getCommitWindowDurationPerc(blockchain);
-                const proofWindowDurationPerc =
-                    await this.blockchainModuleManager.getProofWindowDurationPerc(blockchain);
-                let totalTransactions = await this.calculateTotalTransactions(
-                    blockchain,
-                    commitWindowDurationPerc,
-                    proofWindowDurationPerc,
-                    command.period,
-                );
+        this.operationIdService.emitChangeEvent(
+            OPERATION_ID_STATUS.COMMIT_PROOF.EPOCH_CHECK_START,
+            operationId,
+            blockchain,
+        );
 
-                // We don't expect to have this many transactions in one epoch check window.
-                // This is just to make sure we don't schedule too many commands and block the queue
-                // TODO: find general solution for all commands scheduling blockchain transactions
-                totalTransactions = Math.min(totalTransactions, COMMAND_QUEUE_PARALLELISM * 0.3);
+        const commitWindowDurationPerc =
+            await this.blockchainModuleManager.getCommitWindowDurationPerc(blockchain);
+        const proofWindowDurationPerc =
+            await this.blockchainModuleManager.getProofWindowDurationPerc(blockchain);
+        let totalTransactions = await this.calculateTotalTransactions(
+            blockchain,
+            commitWindowDurationPerc,
+            proofWindowDurationPerc,
+            command.period,
+        );
 
-                const transactionQueueLength =
-                    this.blockchainModuleManager.getTotalTransactionQueueLength(blockchain);
-                if (transactionQueueLength >= totalTransactions) return;
+        // We don't expect to have this many transactions in one epoch check window.
+        // This is just to make sure we don't schedule too many commands and block the queue
+        // TODO: find general solution for all commands scheduling blockchain transactions
+        totalTransactions = Math.min(totalTransactions, COMMAND_QUEUE_PARALLELISM * 0.3);
```

Comment:
This should be moved to epoch check command and passed as parameter (`Math.min(sum(total transactions gnosis, total transactions otp, ...), COMMAND_QUEUE_PARALLELISM * 0.3)`), or divided by number of blockchain implementations

Reply:
good point, thanks @zeroxbt

Reply:
Hi @zeroxbt after reviewing again i don't think that you are right. We did this calculation for each blockchain before this change also. Also why would you do sum(total transaction gnosis, total transactions otp), total number of transactions should be calculated for each blockchain separately.

Reply:
You're right, I think it was wrong before this change also.

This check was added to avoid blocking the command queue. For example, if we have command queue parallelism set to 100, we wanted to avoid having more than 30 commands in total being scheduled by epoch check command.
Currently this check doesn't work as intended in multi chain nodes as you'd have a max of 30 commands being scheduled at max for each chain.

Reply:
You're right that `Math.min(sum(total transactions gnosis, total transactions otp, ...), COMMAND_QUEUE_PARALLELISM * 0.3)` is incorrect. Dividing by number of blockchain implementations is an option, but then if one chain has many txs to be scheduled and the other has 0, the first chain should be allowed to schedule more commands. It's a little harder than i initially thought

Reply:
I see, you have a point. As you said it's a bit complicated to implement this now, and this release should go out asap, so i wouldn't change that part in this PR. This PR should change the way we are scheduling epoch check for each blockchain, but not to change the logic of it. I'll add todo here to be sure we implement this fix in the future.